cmake/onnxruntime_providers.cmake

```diff
@@ -844,28 +866,35 @@ if (onnxruntime_USE_ROCM)
       "${ORTTRAINING_SOURCE_DIR}/training_ops/rocm/*.cc"
     )
 
+    file(GLOB_RECURSE onnxruntime_rocm_training_ops_cu_srcs CONFIGURE_DEPENDS
+      "${ORTTRAINING_SOURCE_DIR}/training_ops/rocm/*.cu"
+      "${ORTTRAINING_SOURCE_DIR}/training_ops/rocm/*.cuh"
+    )
+
+    file(GLOB_RECURSE onnxruntime_rocm_generated_training_ops_cc_srcs CONFIGURE_DEPENDS
+      "${CMAKE_CURRENT_BINARY_DIR}/amdgpu/orttraining/orttraining/training_ops/rocm/*.h"
+      "${CMAKE_CURRENT_BINARY_DIR}/amdgpu/orttraining/orttraining/training_ops/rocm/*.cc"
+    )
+
+    file(GLOB_RECURSE onnxruntime_rocm_generated_training_ops_cu_srcs CONFIGURE_DEPENDS
+      "${CMAKE_CURRENT_BINARY_DIR}/amdgpu/orttraining/orttraining/training_ops/rocm/*.cu"
+      "${CMAKE_CURRENT_BINARY_DIR}/amdgpu/orttraining/orttraining/training_ops/rocm/*.cuh"
+    )
+
     # NCCL is not support in Windows build
-    if (WIN32)
+    if (WIN32 OR NOT onnxruntime_USE_NCCL)
```

Comment:
Isn't `onnxruntime_USE_NCCL` always false on Windows (set in build.py)?  It would be more clear to just check that flag, in my opinion, if possible.

Reply:
as discussed offline, onnxruntime_USE_NCCL is not set in build.py now.